### ISSUE TITLE
Patches do not apply cleanly on Ubuntu 15.04 (3.19.0-18-generic)

### ISSUE THREAD

**Issue by Melab** [rasa/vmware-tools-patches#48]
I'm running Ubuntu 15.04 on Linux 3.19.0-18-generic in VMware **Player** 7.1.0. Upon applying `./patch.sh`, I did not get cleanly applied patches. Here is the output written to the terminal:

    *** Applying vmci/01-vmware9.k3.8rc4.patch ...
    patching file linux/driver.c
    Reversed (or previously applied) patch detected!  Assuming -R.
    Hunk #1 succeeded at 128 with fuzz 1 (offset 4 lines).
    Hunk #2 succeeded at 1745 (offset -5 lines).
    Hunk #3 succeeded at 1976 (offset -2 lines).
    *** Applying vsock/01-vsock-sk_data_ready-kernel-3.15-tools-9.6.2.patch ...
    patching file linux/notify.c
    patching file linux/notifyQState.c
    *** Applying vmblock/01-vmblock.3.10.patch ...
    patching file linux/control.c
    *** Applying vmblock/02-vmblock.3.11.patch ...
    patching file linux/file.c
    *** Applying vmblock/03-vmblock.3.12.patch ...
    patching file linux/control.c
    patching file linux/dentry.c
    patching file linux/inode.c
    *** Applying vmblock/04-vmblock-vfs_readlink-kernel-3.15-tools-9.6.2.patch ...
    patching file linux/inode.c
    Hunk #1 succeeded at 177 (offset -1 lines).
    *** Skipping vmhgfs/01-compat_dcache.h.patch: did not apply cleanly
    *** Skipping vmhgfs/02-patch_inode.c.patch: did not apply cleanly
    *** Applying vmhgfs/03-vmhgfs-3.15-vfs_readlink.patch ...
    patching file link.c
    Hunk #1 succeeded at 160 (offset 9 lines).
    Hunk #2 succeeded at 200 (offset 11 lines).
    *** Applying vmhgfs/04-vmhgfs-d_alias-kernel-3.18.1-tools-9.9.0.patch ...
    patching file inode.c
    patching file Makefile.kernel
    patching file shared/autoconf/d_alias.c
    *** Skipping vmhgfs/05-vmhgfs-d_count-kernel-3.11-tools-9.6.0.patch: did not apply cleanly
    *** Skipping vmhgfs/06-vmhgfs-d_set_type-kernel-3.18.1-tools-9.6.1.patch: did not apply cleanly
    *** Skipping vmhgfs/07-vmhgfs-rename_deprecated-kernel-3.18.1-tools-9.6.1.patch: did not apply cleanly
    *** Applying vmhgfs/08-vmhgfs-smp_mb_before_clear_bit-3.18-tools-9.9.0.patch ...
    patching file page.c
    *** Skipping vmhgfs/09-vmhgfs-uid-gid-kernel-3.12-tools-9.6.1.patch: did not apply cleanly
    *** Skipping vmhgfs/10-vmhgfs-vfs_readlink-kernel-3.16-tools-9.6.2.patch: did not apply cleanly
    *** Skipping vmhgfs/11-vmhgfs-wait_on_bit-kernel-3.17-tools-9.6.2.patch: did not apply cleanly
    *** Skipping vmhgfs/12-vmware9.compat_mm.patch: did not apply cleanly
    *** Applying vmhgfs/13-vmhgfs-f_dentry-kernel-3.19-tools-9.9.2.patch ...
    patching file dir.c
    patching file file.c
    patching file fsutil.c
    patching file shared/compat_dentry.h
    *** Skipping vmhgfs/14-vmhgfs-d_alias-kernel-3.18.1-tools-9.6.2.patch: did not apply cleanly
    *** Applying vmhgfs/15-vmhgfs-bdi-kernel-4.0-tools-9.9.2.patch ...
    patching file filesystem.c
    patching file module.h
    patching file super.c
    *** Skipping vmhgfs/16-vmhgfs-f_dentry-kernel-3.19.0-15-tools-9.6.2.patch: did not apply cleanly

Am I doing it right? Also, supposedly the problem with HGFS was fixed [here](https://github.com/rasa/vmware-tools-patches/issues/29), but I do not think those changes were incorporated into this project.

**Comment by htnmtoi** [comment]
Probably. I get these same issues too, but everything works fine anyway. I assume they're to handle old tools versions; where VMWare has implemented those patches already, the patches will fail.

**Comment by rasa** [comment]
That's true. Not every patch will apply cleanly.

**Comment by htnmtoi** [comment]
I think it would be good to state in README.md that patch failure isn't necessarily a signal of problems. Normally, people assume that patch failure is a bad thing.

**Comment by rasa** [comment]
Even better would be to change the output to simply say something like
`Skipping: patch is not appropriate for this kernel`

**Comment by htnmtoi** [comment]
I agree.

**Comment by jasperf** [comment]
I got a `Skipping vmhgfs/15-vmhgfs-bdi-kernel-4.0-tools-9.9.2.patch: patch not appropriate for this kernel`

`cat /proc/version
Linux version 3.13.0-61-generic (buildd@lgw01-50) (gcc version 4.8.2 (Ubuntu 4.8.2-19ubuntu1) ) #100-Ubuntu SMP Wed Jul 29 11:21:34 UTC 2015`

 and therefore cannot load items from a shared folder. Updating the repo did not help. I guess I will have to wait for the repo to be updated? Perhaps that could be added to the message as well

**Comment by jasperf** [comment]
I got a `Skipping vmhgfs/15-vmhgfs-bdi-kernel-4.0-tools-9.9.2.patch: patch not appropriate for this kernel`

`cat /proc/version
Linux version 3.13.0-61-generic (buildd@lgw01-50) (gcc version 4.8.2 (Ubuntu 4.8.2-19ubuntu1) ) #100-Ubuntu SMP Wed Jul 29 11:21:34 UTC 2015`

 and therefore cannot load items from a shared folder. Updating the repo did not help. I guess I will have to wait for the repo to be updated? Perhaps that could be added to the message as well.
Funnily enough when I ran regular VMWare Tools again all ran well and sharing got working again and I was able to access the attached folder loading my DropBox data..
